onnxruntime/core/providers/dnnl/subgraph/dnnl_convgrad.h

```diff
@@ -314,33 +312,50 @@ class DnnlConvGrad : public DnnlKernel {
             *conv_bwd_weights_desc_, engine_to_use, *(conv_fwd_->GetPrimitiveDesc())));
 
     diff_dst_mem_ = onnxruntime::make_unique<dnnl::memory>(
-        dnnl::memory(conv_bwd_weights_pd_.get()->diff_dst_desc(), engine_to_use, nullptr));
+        dnnl::memory(conv_bwd_weights_pd_.get()->diff_dst_desc(), cpu_engine, nullptr));
 
     src_mem_ = onnxruntime::make_unique<dnnl::memory>(
-        dnnl::memory(conv_bwd_weights_pd_.get()->src_desc(), engine_to_use, nullptr));
+        dnnl::memory(conv_bwd_weights_pd_.get()->src_desc(), cpu_engine, nullptr));
```

Comment:
Style nit: for all of these `make_unique` calls, I think you shouldn't need to write the type name type. e.g. The following should work:

```c++
src_mem_ = onnxruntime::make_unique<dnnl::memory>(conv_bwd_weights_pd_.get()->src_desc(), cpu_engine, nullptr);
```

Reply:
@mrry This is one of those things that stuck around because I was following the style of past developers.  I looked at the code in other files, for this endpoint and this style nit is everywhere. I am already working on a cleanup commit that only does code cleanup.  This would be better addressed in that commit instead of this commit.

Reply:
Sounds great, thanks.